### PR TITLE
[12.0][ADD] (sale)_account_taxjar: TaxJar Odoo Integration for sale and invoicing

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Account TaxJar
--------------------

This module allow to automate your sales taxes on invoices calculation using
`TaxJar SmartCalcs API <https://www.taxjar.com/smartcalcs/>`_.

This module does:

 * Import/update TaxJar product categories.
 * Import/update TaxJar Nexus.
 * Request for a Tax Rate calculation.

This module does't:

 * Manage transactions with TaxJar Dashboard
 * Manage customers with TaxJar Dashboard
 * Validate addresses

Sale Account TaxJar
---------------------------

This module extends the functionality of `account_taxjar` to sales orders.

CC ~ @Eficent

### CODE REVIEW

**Comment by HviorForgeFlow** [comment]
This needs revisions @mreficent @lreficent @OCA/local-usa-maintainers

**Comment by max3903** [comment]
@hveficent Can you split this pull request by module? 58 new files is a lot to review.

**Comment by HviorForgeFlow** [comment]
@max3903 I wish I won't, functional review is easier this way.

File: account_taxjar/models/product.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2018-2019 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    tax_code_id = fields.Many2one('taxjar.category', string="TaxJar Tax Code")
```
Comment (StefanRijnhart) [inline account_taxjar/models/product.py:10]: This column name is very generic. Would you consider renaming to taxjar_code_id?

**Comment by StefanRijnhart** [comment]
Thank you @HviorForgeFlow. This PR looks good in general. We wanted to propose a port of our custom 8.0 taxjar module for version 12.0, but you beat us to it. We would like to add additional features though:

* Address validation
* Pass the invoiced shipping amount separately as per the Taxjar API
* Create a salestax transaction in Taxjar on invoice confirmation, in case Taxjar handles the declarations (using the asynchronous job queue)

Should we try and get this merged first, and then proceed with the additional features? Should we put the declaration bit be in a separate module so that we don't add the additional dependency on queue_job, or should we integrate it?

**Comment by HviorForgeFlow** [comment]
Hi @StefanRijnhart, actually we are not using this module anymore, so if you are planning to reuse this code I will appreciate it. Feel free to merge/super seed it adding features you propose or replacing the whole module. I will review it.

Yes, maybe queue_job integration could be managed in a separated module if there aren't major inconveniences.

**Comment by StefanRijnhart** [comment]
Oh, that's sad to hear. We were hoping to start a little taxjar user group here ;-) Are you using anything else instead, or talking to another tax provider?

**Comment by StefanRijnhart** [comment]
OK, all things considered we already have a version 12.0 of our 8.0 production code, while the code of this PR is not actually in use anywhere so we will propose our own version (which has a very different setup).

**Pull request closed without merging**